privacy-protections/storage-blocking/helpers/commonTests.js

```diff
@@ -1,7 +1,53 @@
 /* exported commonTests */
-/* global cookieStore */
+/* global cookieStore, THIRD_PARTY_TRACKER_ORIGIN, THIRD_PARTY_ORIGIN */
+
 // tests that are common for both main frame and an iframe
 const commonTests = [
+    {
+        id: 'first party header cookie',
+        store: (data) => {
+            return fetch(`/set-cookie?value=${data}`).then(r => {
+                if (!r.ok) {
+                    throw new Error('Request failed.');
+                }
+            });
+        },
+        retrive: () => {
+            return fetch('/reflect-headers')
+                .then(r => r.json())
+                .then(data => data.headers.cookie.match(/headerdata=([0-9]+)/)[1]);
+        }
+    },
+    {
+        id: 'safe third party header cookie',
```

Comment:
hm, it's a bit awkward that in iframes this and next test are redundant (`${THIRD_PARTY_ORIGIN}/set-cookie` === `/set-cookie`). They are also not third party in this context so the description is wrong. Can we somehow elegantly exclude them from running in frames where origins match?